tools/arcilator/CMakeLists.txt

```diff
@@ -15,6 +15,7 @@ target_link_libraries(arcilator
   CIRCTSupport
   CIRCTTransforms
   MLIRBuiltinToLLVMIRTranslation
+  MLIRExecutionEngine
```

Comment:
I know that this PR is still a draft, but just wanted to make you aware that the MLIRExecutionEngine is only conditionally built IIRC when the host machine architecture is supported (and can also be manually disabled) which means we would need to make arcilator conditional on this as well. However, arcilator could also be useful to build the simulation binary on one machine and run it on another one (with potentially different architecture), so maybe we can always build arcilator but use some `#ifdef` in the code to only make the JIT part conditional? Not exactly sure what's the best approach here, maybe just guarding whole arcilator is good enough for a start.